src/Plugin/GraphQL/DataProducer/Entity/Fields/Image/ImageDerivative.php

```diff
@@ -67,14 +120,17 @@ public function resolve(FileInterface $entity = NULL, $style, RefinableCacheable
       $image_style->transformDimensions($dimensions, $entity->getFileUri());
       $metadata->addCacheableDependency($image_style);
 
-      return [
-        'url' => $image_style->buildUrl($entity->getFileUri()),
-        'width' => $dimensions['width'],
-        'height' => $dimensions['height'],
-      ];
+      $context = new RenderContext();
```

Comment:
Please add a code comment why we need the RenderContext here. Something like "// Prevent 'leaked metadata found, check if rendered too early' errors. @todo Find out why the render system is invoked at all?"